tests/Integration/Behaviour/Features/Scenario/Customer/customer_management.feature

```diff
@@ -79,3 +79,37 @@ Feature: Customer Management
       | groupIds       | [Guest]                   |
     And I delete customer "CUST-5" with "allow registration after deletion" checked
     Then the customer "CUST-5" should not be found
+
+  Scenario: Edit guest customer email
+    When I create a customer "CUST-6" with following properties:
+      | firstName      | Mathieu                     |
+      | lastName       | Guest                       |
+      | email          | guest@prestashop.com        |
+      | password       | PrestaShopForever1_!        |
+      | defaultGroupId | Guest                       |
+      | groupIds       | [Guest]                     |
+      | isGuest        | true                        |
```

Comment:
@Progi1984 @atomiix Guys can you help me get isGuest here? I need it for this test to work properly. :(

Reply:
Actually the command Handler `src/Adapter/Customer/CommandHandler/AddCustomerHandler.php doesn't manage guest

Reply:
Any idea @matks ?

Reply:
> Actually the command Handler `src/Adapter/Customer/CommandHandler/AddCustomerHandler.php doesn't manage guest

Well, that's expected 😄  as the name says this is a handler that creates Customers, not Guests. Maybe we should have a different Command for Guests? If there are main differences between a Guest and a Customer (and I think "having a password" and thus "be able to login" is a main difference) we could have different handlers for the two. It's very different to create a new guest and to create a new customer. As @Hlavtox mentioned: "a guest can have any email it wants, which means you can have fifty guests with same emails" while "a customer email must be unique" so it's really 2 different business usecases.

I think that, in order to implement this logic correctly, we should first have a look at the global concept of guests in PrestaShop (sorry, I know it was not the initial scope of the PR, but I think without this we might be monkey-patching a root issue and make it worse).

What can guests do in PrestaShop?
- can they have multiple orders?
- can they be edited? can they be deleted? can they be bulk deleted? can they be disabled? can they be bulk disabled?
- can an invoice refer to a guest?
- what happens when a guest is transformed into a customer?

Depending on the answers to these questions, we may be able to understand "are guests very similar to customers but there is a few different things" or "guests and customers are very different".

If guests are very similar to customers we should be able to rely on `if ($customer->is_guest)` for the differences. We should only have a few of them, like an option.

If guests are very different to customers, this means that if we start using `if ($customer->is_guest)` we'll very soon have ten, twenty, thirty of these `if ($customer->is_guest)` which will double, triple the number of independent code paths of our code and skyrocket our [cyclomatic complexity](https://en.wikipedia.org/wiki/Cyclomatic_complexity) and this is something that usually is bad for software. In this usecase, it might be better to have new commands and handlers: AddGuestHander, EditGuestHandler, DeleteGuestHandler ...

Reply:
@matks While I understand all your concerns, I think its better if we leave it like this, because it just WORKS. 😄

The differences between guest and non-guest are not that big that it couldn't be tackled with some simple conditions.

## Key data difference between customer and guests
- Guest has is_guest = 1
- Guest should have default group set as PS_GUEST_GROUP from prestahop configuration.
- Guest should belong to only one group - PS_GUEST_GROUP from prestahop configuration.